### ISSUE TITLE
ansible-builder failed to find requirements.txt

### ISSUE THREAD

**Issue by lamm** [theforeman/foreman-ansible-modules#1465]
##### SUMMARY
ansible-builder failed to build EE when using theforeman.foreman ansible collection in requirements.yml

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION
N/A

##### COLLECTION VERSION
3.5.0

##### KATELLO/FOREMAN VERSION
N/A

##### STEPS TO REPRODUCE

execution-environment.yml
```yaml
---
version: 1

build_arg_defaults:
  EE_BASE_IMAGE: 'quay.io/ansible/ansible-runner:stable-2.12-latest'
  ANSIBLE_GALAXY_CLI_COLLECTION_OPTS: -vvv

dependencies:
  galaxy: requirements.yml
```
requirements.yml
```yaml
---
collections:
  - name: ansible.netcommon
  - name: theforeman.foreman
```

##### EXPECTED RESULTS
Ansible Execution Environment image to be build.

##### ACTUAL RESULTS
ansible-builder failed to find requirements.txt file.

```
ansible-builder build

Running command:
  podman build -f context/Containerfile context
...showing last 20 lines of output...
[1/3] STEP 6/7: RUN ansible-galaxy role install -r requirements.yml --roles-path "/usr/share/ansible/roles"
--> Using cache 5e9c0f889b01ce89f5ae2e734a93d5fbae716408de039bae8146f5eb0b0fc2bc
--> 5e9c0f889b0
[1/3] STEP 7/7: RUN ANSIBLE_GALAXY_DISABLE_GPG_VERIFY=1 ansible-galaxy collection install $ANSIBLE_GALAXY_CLI_COLLECTION_OPTS -r requirements.yml --collections-path "/usr/share/ansible/collections"
--> Using cache 14d6a96e8694df0a65cc3ce4acc110fd3f46e71d22368275fb3d138e23b35d7a
--> 14d6a96e869
[2/3] STEP 1/6: FROM quay.io/ansible/ansible-builder:latest AS builder
[2/3] STEP 2/6: COPY --from=galaxy /usr/share/ansible /usr/share/ansible
--> Using cache fbd7d9e2ef4ef5eb1e49c484ef1305816504cc7b682f67906c7824fe7eda11d4
--> fbd7d9e2ef4
[2/3] STEP 3/6: ADD _build/requirements.txt requirements.txt
--> Using cache 996ad0aa40fb9b967c1715115a9d3d1fd56aa7c7b13502062d02168fcb8c4fea
--> 996ad0aa40f
[2/3] STEP 4/6: ADD _build/bindep.txt bindep.txt
--> Using cache 0800e34f96c2b38005858d006212b88159b57ebe888ee5d4ce2bdc0360b83f0b
--> 0800e34f96c
[2/3] STEP 5/6: RUN ansible-builder introspect --sanitize --user-pip=requirements.txt --user-bindep=bindep.txt --write-bindep=/tmp/src/bindep.txt --write-pip=/tmp/src/requirements.txt
Expected requirements file not present at: /usr/share/ansible/collections/ansible_collections/theforeman/requirements.txt
FileNotFoundError: [Errno 2] No such file or directory: '/usr/share/ansible/collections/ansible_collections/theforeman/foreman/../requirements.txt'
Error: error building at STEP "RUN ansible-builder introspect --sanitize --user-pip=requirements.txt --user-bindep=bindep.txt --write-bindep=/tmp/src/bindep.txt --write-pip=/tmp/src/requirements.txt": error while running runtime: exit status 1

An error occured (rc=1), see output line(s) above for details.
```

**Comment by lamm** [comment]
The issue seems to be located in meta/execution-environment.yml file.

```yaml
---
version: 1

build_arg_defaults:
  EE_BASE_IMAGE: 'quay.io/ansible/ansible-runner:latest'

dependencies:
  python: ../requirements.txt
  system: ../bindep.txt
```
When removing relative path on dependencies like this :
```yaml
---
version: 1

build_arg_defaults:
  EE_BASE_IMAGE: 'quay.io/ansible/ansible-runner:latest'

dependencies:
  python: requirements.txt
  system: bindep.txt
```
then image build successfully.

**Comment by evgeni** [comment]
When we initially added EE metadata, we had the paths like you, and then it wouldn't find it when building the EE itself in
https://github.com/theforeman/foreman-ansible-modules/blob/1e8f5a818848f92ca38f9fa97c3b4737624ad750/.github/workflows/main.yml#L140

We then talked to the builder team and they confirmed `../path` would be correct.

Can you please bring this example to the builder team and ask them for proper docs how the metadata needs to refer to dependencies?

**Comment by evgeni** [comment]
When we initially added EE metadata, we had the paths like you, and then it wouldn't find it when building the EE itself in
https://github.com/theforeman/foreman-ansible-modules/blob/1e8f5a818848f92ca38f9fa97c3b4737624ad750/.github/workflows/main.yml#L140

We then talked to the builder team and they confirmed `../path` would be correct.

Can you please bring this example to the builder team and ask them for proper docs how the metadata needs to refer to dependencies?

**Comment by lamm** [comment]
A simillar issue was already opened : https://github.com/ansible/ansible-builder/issues/393
Answer from buider team was :
- remove relative pathing from meta/execution-environment.yml to make `ansible-builder introspect` working correcly
- Use a dedicated file for `ansible-builder builder`

**Comment by evgeni** [comment]
Ah, very interesting. Thanks for the pointers!

**Comment by evgeni** [comment]
This was fixed in #1471
